tests/cases.py

```diff
@@ -642,8 +640,8 @@ def test_reset(self) -> None:
     def test_update(self) -> None:
         """Test method `update`."""
         # Generate random tensors
-        a = torch.rand(self.batch_size, 10, device=self.device, generator=self.generator)
-        b = torch.rand(self.batch_size, 20, device=self.device, generator=self.generator)
+        a = torch.rand(self.batch_size, 10, generator=self.generator).to(device=self.device)
```

Comment:
@cthoyt this pattern occurs frequently for tests - maybe we should create a new helper method for this?

Reply:
some background: since we explcitly initialize the random number generator and save it to `self.generator` before we know about the device, the generator is a cpu-only generator. Sadly, we cannot simply move it to the device, but rather have to create the tensor on cpu and then move them to device.

Reply:
I think I looked into this before - it looks like the generator doesn't have support for device choice :/

Reply:
Done in 0d38b560. There are other functions in `pykeen.utils` that are only used in testing, we could also consider moving those over in a future PR